paymentsheet/src/main/java/com/stripe/android/paymentelement/confirmation/DefaultConfirmationHandler.kt

```diff
@@ -86,11 +87,7 @@ internal class DefaultConfirmationHandler(
             return
         }
 
-        _state.value = ConfirmationHandler.State.Confirming(arguments.confirmationOption)
```

Comment:
We were relying on the behavior of setting the value before launching the confirm. But I didn't want to duplicate it. So now we rely solely on the one in confirm, but I changed some threading up in confirm to support the old behavior that CustomerSheet was relying on.